xlb/operator/boundary_masker/mesh_boundary_masker.py

```diff
@@ -59,67 +58,51 @@ def _construct_warp(self):
         @wp.kernel
         def kernel(
             mesh_id: wp.uint64,
-            origin: wp.vec3,
-            spacing: wp.vec3,
             id_number: wp.int32,
             bc_mask: wp.array4d(dtype=wp.uint8),
             missing_mask: wp.array4d(dtype=wp.bool),
-            start_index: wp.vec3i,
         ):
             # get index
             i, j, k = wp.tid()
 
             # Get local indices
-            index = wp.vec3i()
-            index[0] = i - start_index[0]
-            index[1] = j - start_index[1]
-            index[2] = k - start_index[2]
+            index = wp.vec3i(i, j, k)
 
             # position of the point
             ijk = wp.vec3(wp.float32(index[0]), wp.float32(index[1]), wp.float32(index[2]))
-            ijk = ijk + wp.vec3(0.5, 0.5, 0.5)  # cell center
-            pos = wp.cw_mul(ijk, spacing) + origin
-
+            pos = ijk + wp.vec3(0.5, 0.5, 0.5)  # cell center
             # Compute the maximum length
-            max_length = wp.sqrt(
-                (spacing[0] * wp.float32(bc_mask.shape[1])) ** 2.0
-                + (spacing[1] * wp.float32(bc_mask.shape[2])) ** 2.0
-                + (spacing[2] * wp.float32(bc_mask.shape[3])) ** 2.0
-            )
+            max_length = wp.sqrt(2.0) / 2.0  # half of unit cell diagonal
```

Comment:
I like these clean ups but please check and raise some sort of error if spacing is not 1.

Reply:
There is no spacing. It is fully removed. The assumption is that the mesh is prepared inside the example. Not inside the library.

Reply:
Here we simply check if there is a mesh in each direction up to that max_length (which is now a constant). The results are identical and all you need to do now is to resize the mesh by the dimensions that fit inside your domain inside the example file (which makes sense to do, just a simple division by dx, instead of passing multiple parameters to the kernel).

Reply:
I know what has been done. I am just saying we should somehow check if the user has prepared the mesh correctly and raise errors if not.

Reply:
I will see if we can add this in the masker constructor. We can't do very rigor check (since you never know what the user intention is) but we change check the bounds of the mesh.